python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1116,6 +1291,13 @@ def _initialize_adapter(self, args: Sequence[str]) -> BaseAdapter:
         project = load_project(self.project_dir, False, profile, {})
         config = RuntimeConfig.from_parts(project, profile, flags)
 
+        from dbt.adapters.duckdb.credentials import DuckDBCredentials
+
+        if isinstance(config.credentials, DuckDBCredentials):
+            if not config.credentials.config_options:
+                config.credentials.config_options = {}
+            config.credentials.config_options["access_mode"] = "READ_ONLY"
```

Comment:
Could you explain what happening here?

Reply:
attempts to get duckdb adapter to play nice - concurrently, duckdb only accepts either:
a) one read-write connection
b) many read-only connections

this lets us parallelize our accesses for postprocessing in the duckdb case - sadly don't think we can run any of these tasks simultaneously to the dbt build since that's read-write